tests/test_plugin_loader_runner.py

```
@@ -0,0 +1,93 @@
+
+    def test_run_type_error_non_string_args(self):
+        """Test _run raises TypeError for non-string arguments."""
+        with pytest.raises(TypeError) as excinfo:
+            _run(["git", 123])  # type: ignore[list-item]
+        self.assertIn("all command arguments must be strings", str(excinfo.value))
+
+    def test_run_value_error_shell_true(self):
+        """Test _run raises ValueError for shell=True."""
+        with pytest.raises(ValueError) as excinfo:
+            shell_flag = True
+            _run(["git", "status"], shell=shell_flag)  # nosec B604
```

Ruff (0.15.12)

[warning] `pytest.raises()` block should contain a single simple statement

(PT012)